src/lib/server/storages/health.ts

```
@@ -0,0 +1,121 @@
+    }
+}
+
+
+/**
+ * Check the health of a single storage and update its status if necessary.
+ * @param storage The storage object to check.
+ * @return The updated storage object
+ */
+export async function updateStorageHealth(storage: typeof storages.$inferSelect) {
+    const [ locks, desyncedBackups, staleSnapshots ] = await Promise.all([
+        getRepositoryLocks(storage.url, storage.password!, storage.env),
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)